specifications/Dutch Auction.md

```diff
@@ -0,0 +1,172 @@
+
+#### Goals
+Establish a mechanism to determine the price to become a validator.
+
+#### Non Goals
+The mechanism to become a validator.
+
+#### Assumptions
+Values defined at deployment are defined as a guide and can be modified to adapt the price curve to required results.
+
+## Specification
+
+#### Overview
+The solution consists of a Smart Contract that provides the following functionality:
+##### Start an auction
+This operation allows to start an auction by calculating the initial and final prices and registering the current block number as auction's start block, upon execution, an event with all auction details will be emitted.
+##### Get current bidding price
+This operation calculates the current bidding price using a specific price curve and the number of blocks mined between current block and the registered start block.
+##### Bid for current auction price
+This operation emits an event with address of the winner (sender) and auction details
+
+#### Data
+These variables will govern the price curve, the values are set at deployment
+| Descriptor | Constructor Value |
+| ----------- | ----------- |
+| Start Price  | 1000000 * 10 ** 18 |
+| Decay | 16 |
+| Scale Parameter | 100 |
+
+The cost to add a validator is calculated as follows:
+* ETHDKG Single Validator Cost (Two ETHDKG units operations 1.2M gas units) -> 1200000 * 2 * gasPriceInWeis
+
+Auction's final price can be obtained multiplying the current number of registered validators by ETHDKG Single Validator Cost.
+
+#### Logic
+##### Start Auction
+This function will be called when a position for a new validator is open, and performs the following actions:
+* Defines auction final price multiplying ETHDKG Validator Cost by the number of current validators in network (The bidding price will never be lower than this value)
+* Determines the current auction id by increasing a counter.
+* Defines the auction's start block as the current block number.
+* Defines initial and final price
+* Upon execution emit an AuctionStarted event with auction id, initial price and final price
+This operation can only be executed by factory
+
+##### Get Bid Price
+This function will be called when a bidder wants to know the current bid price, and performs the following actions: 
+* Calculates the number of blocks between current block and auction's start block (time elapsed since start).
+* Determines current auction price using the following function:
+```solidity
+    function _dutchAuctionPrice(uint256 blocks) internal view returns (uint256 result) {
+        uint256 _alfa = _startPrice - _finalPrice;
+        uint256 t1 = _alfa * _scaleParameter;
+        uint256 t2 = _decay * blocks + _scaleParameter ** 2;
+        uint256 ratio = t1 / t2;
+        return _finalPrice + ratio;
+    }
+```
+This operation is public
+
+##### Bid for Current Price
+This function will be called when a bidder wants to buy for the bidding price, and performs the following actions: 
```

Comment:
Will this bid cause the auction to close, and therefore block any other future bids?

Reply:
Yes, the initial requirement only supports one opening. Added blocking of future bids by closing the auction.